tests/NLog.UnitTests/Targets/MailTargetTests.cs

```diff
@@ -695,19 +736,22 @@ public MockSmtpClient()
                 this.MessagesSent = new List<MailMessage>();
             }
 
+            public SmtpDeliveryMethod DeliveryMethod { get; set; }
             public string Host { get; set; }
             public int Port { get; set; }
             public int Timeout { get; set; }
+            public string PickupDirectoryLocation { get; set; }
+
 
             public ICredentialsByHost Credentials { get; set; }
             public bool EnableSsl { get; set; }
             public List<MailMessage> MessagesSent { get; private set; }
 
-            public void Send(MailMessage msg)
+            public new void Send(MailMessage msg)
             {
-                if (string.IsNullOrEmpty(this.Host))
+                if (string.IsNullOrEmpty(this.Host) && string.IsNullOrEmpty(this.PickupDirectoryLocation))
```

Comment:
@304NotModified the `if` is checking for the scenario were both are null or empty, is not demanding that both are present.